rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/model/ParametricSummaryBridge.java

```diff
@@ -346,29 +357,55 @@ public CompletableFuture<List<Location>> getImplementations(Position cursor) {
         return implementations.thenCompose(d -> d.lookup(cursor).get());
     }
 
-    public void calculateSummary() {
-        calculateSummary(false);
+
+    public class SummaryCalculation {
+        public final CompletableFuture<VersionedTree> input;
```

Comment:
i'm not a fan of the names `input` and `output`. `tree` and `messages` would be better in my opinion. and the `calculation` is actually the result of it, so maybe `summary` would be a better name?

Reply:
Fixed